db/local_db/start_database.go

```diff
@@ -221,11 +228,33 @@ func startPostgresProcess(port int, listen StartListenType, invoker constants.In
 		Foreground: false,
 	}
 
-	err := postgresCmd.Start()
+	stopCollectionFn, err := setupLogCollector(postgresCmd, logChannel)
+	if err != nil {
+		return err
+	}
+	defer stopCollectionFn()
+
+	err = postgresCmd.Start()
 	if err != nil {
 		return err
 	}
 
+	go func() {
```

Comment:
this is already a massive function - factor this into a separate function
```
go displayServiceLogs()
```

Reply:
fixed